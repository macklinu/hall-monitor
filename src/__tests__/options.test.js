import hallMonitor from '..'

const rule = context => {
  const options = context.options[0] || []
  return {
    CallExpression(node) {
      if (options.includes(node.callee.name)) {
        context.report({
          node,
          message: "No call expressions named '{{ name }}'",
          data: { name: node.callee.name },
        })
      }
    },
  }
}

hallMonitor('options', rule, {
  valid: [''],
  invalid: [{ code: 'badFunctionName()', options: ['badFunctionName'] }],
})
