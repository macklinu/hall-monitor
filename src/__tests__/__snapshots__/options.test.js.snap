// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options invalid badFunctionName() 1`] = `
"badFunctionName()
 ^^^^^^^^^^^^^^^^^ [No call expressions named 'badFunctionName']"
`;

exports[`options valid [no code] 1`] = `""`;
